### PR TITLE
membership: ensure parity with active and passive members

### PR DESCRIPTION
**Pull request by FintanH**
When we add active we never checked to remove them from the passive
list. This means we end up double counting a peer.

Signed-off-by: Fintan Halpenny <fintan.halpenny@gmail.com>

### CODE REVIEW

**Comment by FintanH** [comment]
One thing I'm thinking is that a `promote` function might make more sense, but without being too familiar I'd have to scope around the usages a bit to understand it better. Putting this up now for early feedback

**Review by xla — DISMISSED** [review]
（╹◡╹）

**Comment by kim** [comment]
> One thing I'm thinking is that a `promote` function might make more sense

There used to be one I think, but the only difference was that promotion implies that it was passive before, while `add_active` doesn't. So this is just splitting hairs on words, which I'm known to react to in grumpy ways.

File: librad/src/net/protocol/membership/partial_view.rs [hunk 1]
```diff
@@ -127,12 +127,16 @@ where
...
 
-        let demoted = if self.num_active() >= self.max_active {
+        let demoted = if self.is_active_full() {
             self.demote_random()
         } else {
             vec![]
         };
 
+        if self.is_passive(&info.peer_id) {
+            self.passive.remove(&info.peer_id);
```
Comment (kim) [inline librad/src/net/protocol/membership/partial_view.rs:137]: This needs to add an `Evicted` transition to the return value, for example by calling `evict`.

**Review by kim — CHANGES_REQUESTED** [review]

**Review by kim — DISMISSED** [review]
:ok_hand:

**Pull request merged**